### PR TITLE
MQTT State engine refactor and hygiene

### PR DESCRIPTION
**Pull request by muneebahmed10**
*Description of changes:*
* Refactor some state engine functions to return status codes.
* Reduce complexity for some MQTT functions.
* Add strerror utility for MQTT status codes and state engine states.
* Fix some build warnings for the MQTT library.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: libraries/standard/mqtt/src/mqtt.c [hunk 1]
```diff
@@ -283,14 +292,11 @@ static MQTTPubAckType_t getAckFromPacketType( uint8_t packetType )
             break;
 
         case MQTT_PACKET_TYPE_PUBCOMP:
-            ackType = MQTTPubcomp;
-            break;
-
         default:
-
             /* This function is only called after checking the type is one of
```
Comment (abhidixi11) [inline libraries/standard/mqtt/src/mqtt.c:296]: Update the comment here to reflect the change.

Comment (muneebahmed10) [reply libraries/standard/mqtt/src/mqtt.c:296]: Updated

File: libraries/standard/mqtt/include/mqtt_state.h [hunk 2]
```diff
@@ -118,4 +123,13 @@ uint16_t MQTT_StateSelect( const MQTTContext_t * pMqttContext,
                            MQTTPublishState_t searchState,
                            MQTTStateCursor_t * pCursor );
 
+/**
+ * @brief Error code to string conversion for state engine.
```
Comment (aggarg) [inline libraries/standard/mqtt/include/mqtt_state.h:127]: Is it error code or publish state i.e. Publish State to string conversion for state engine.

**Pull request merged**